File: src/main/java/com/m2t/hachikodictionary/model/LearnedWord.java
package com.m2t.hachikodictionary.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("learned_words")
public class LearnedWord {

    @Id
    private Account account;

    private Word word;

    private int level;
}

File: src/main/java/com/m2t/hachikodictionary/model/LearnedWord.kt
package com.m2t.hachikodictionary.model

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.IdClass
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import org.hibernate.annotations.GenericGenerator

@Entity
@Table(name = "learned_words")
@IdClass(LearnedWordId::class)
data class LearnedWord(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id", nullable = false)
    @Id
    val account: Account,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "word_id", nullable = false)
    @Id
    val word: Word,

    var level: Int,
){
}